internal/impl/devbox.go

```diff
@@ -953,8 +953,10 @@ func (d *Devbox) ensurePackagesAreInstalledAndComputeEnv(
 
 	// When ensurePackagesAreInstalled is called with ensure=true, it always
 	// returns early if the lockfile is up to date. So we don't need to check here
-	if err := d.ensurePackagesAreInstalled(ctx, ensure); err != nil {
+	if err := d.ensurePackagesAreInstalled(ctx, ensure); err != nil && !strings.Contains(err.Error(), "no such host") {
 		return nil, err
+	} else if err != nil {
+		ux.Fwarning(d.stderr, "Error connecting to the internet. Will attempt to use cached environment.\n")
```

Comment:
I guess there is some danger that `ensurePackagesAreInstalled` has done some work already (like nix profile install) but then internet fails and we fallback to using a stale `print-dev-env` cache value

Reply:
(i think its okay; better than blocking)

Reply:
What kind of things would be stale?

Reply:
like it may have some package that has been removed, or an older version of a package, etc.

Reply:
Usually, this means one of 2 things:

* Something in devbox.json changed
* devbox version changed (In this case we bust the cache because the .devbox dir structure might change)

The cached environment is nix only. So if you add a script or env var to your `devbox.json` we will still pick it up. If you add a package we will not.

So one edge case where this doesn't completely work is if you add a local flake. There's no internet needed to install it, but it won't work anyway. This is harder to fix given our current code structure and feels pretty edge case so I wouldn't worry.